src/Anyt/BugTrackerBundle/Tests/Unit/EventListener/NoteListenerTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+namespace Anyt\BugTrackerBundle\Tests\Unit\EventListener;
+
+use Anyt\BugTrackerBundle\Entity\Issue;
+use Anyt\BugTrackerBundle\EventListener\NoteListener;
+use Doctrine\Common\Util\ClassUtils;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Oro\Bundle\NoteBundle\Entity\Note;
+
+class NoteListenerTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var NoteListener
+     */
+    protected $listener;
+
+    /**
+     * @var OnFlushEventArgs|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $args;
+
+    /**
+     * @var \Doctrine\ORM\UnitOfWork|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $uow;
+
+    /**
+     * @var \Doctrine\ORM\EntityManager|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $em;
+
+    /**
+     * @var Note
+     */
+    protected $entity;
+
+    /**
+     * @var Issue
+     */
+    protected $targetEntity;
+
+    public function setUp()
+    {
+
+        $this->listener = new NoteListener();
+
+        $this->entity = $this->getMock('Oro\Bundle\NoteBundle\Entity\Note');
+
+        $this->targetEntity = new Issue();
+        $this->entity->method('getTarget')->willReturn($this->targetEntity);
+
+        $this->args = $this->getMockBuilder('\Doctrine\ORM\Event\OnFlushEventArgs')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->em = $this->getMockBuilder('\Doctrine\ORM\EntityManager')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->uow = $this->getMockBuilder('\Doctrine\ORM\UnitOfWork')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+
+        $this->em->method('getUnitOfWork')->willReturn($this->uow);
+
+        $metadataFactory = $this->getMock('\Doctrine\ORM\Mapping\ClassMetadataFactory');
+        $classMetadata = $this->getMockBuilder('\Doctrine\ORM\Mapping\ClassMetadata')
+            ->disableOriginalConstructor()->getMock();
+        $metadataFactory->method('getMetadataFor')->willReturn($classMetadata);
+
+        $this->em->method('getMetadataFactory')->willReturn($metadataFactory);
+
+        $this->args->method('getEntityManager')->willReturn($this->em);
+
+    }
+
+    public function testOnFlush()
```

Comment:
Please check that date was set only once
Please check different entities
Please check that entity is not Note
Please check that target entity is not Issue

please add expects on uow with recompute methods